static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -0,0 +1,413 @@
+          setEditingThresholds(updatedEditingThresholds);
+        });
+    });
+  };
+
+  const deleteThreshold = thresholdId => {
+    const updatedEditingThresholds = {...editingThresholds};
+    const thresholdData = editingThresholds[thresholdId];
+    const path = `/projects/${orgSlug}/${thresholdData.project.slug}/release-thresholds/${thresholdId}/`;
+    const method = 'DELETE';
+    if (!thresholdId.includes(NEW_THRESHOLD_PREFIX)) {
+      const request = api.requestPromise(path, {
+        method,
+      });
+      request
+        .then(() => {
+          refetch();
+        })
+        .catch(_err => {
+          setError('Issue deleting threshold');
+          const errorThreshold = {
+            ...thresholdData,
+            hasError: true,
+          };
+          updatedEditingThresholds[thresholdId] = errorThreshold as EditingThreshold;
+          setEditingThresholds(updatedEditingThresholds);
+        });
+    }
+    delete updatedEditingThresholds[thresholdId];
+    setEditingThresholds(updatedEditingThresholds);
+  };
+
+  const closeEditForm = thresholdId => {
+    const updatedEditingThresholds = {...editingThresholds};
+    delete updatedEditingThresholds[thresholdId];
+    setEditingThresholds(updatedEditingThresholds);
+  };
+
+  const editThresholdState = (thresholdId, key, value) => {
+    if (editingThresholds[thresholdId]) {
+      const updateEditing = JSON.parse(JSON.stringify(editingThresholds));
+      updateEditing[thresholdId][key] = value;
+      setEditingThresholds(updateEditing);
+    }
+  };
+
+  return (
+    <StyledThresholdGroup columns={columns}>
+      {Array.from(thresholdIdSet).map((tId: string, idx: number) => {
+        const threshold = editingThresholds[tId] || thresholdsById[tId];
+        return (
+          <StyledRow
+            key={threshold.id}
+            lastRow={idx === thresholdIdSet.size - 1}
+            hasError={threshold.hasError}
+          >
+            <FlexCenter style={{borderBottom: 0}}>
+              {idx === 0 ? (
+                <ProjectBadge
+                  project={threshold.project}
+                  avatarSize={16}
+                  hideOverflow
+                  disableLink
+                />
+              ) : (
+                ''
+              )}
+            </FlexCenter>
+            <FlexCenter style={{borderBottom: 0}}>
+              {idx === 0 ? threshold.environment.name || 'None' : ''}
+            </FlexCenter>
+            {/* FOLLOWING COLUMNS ARE EDITABLE */}
+            {editingThresholds[threshold.id] ? (
+              <Fragment>
+                <FlexCenter>
+                  <Input
+                    style={{width: '50%'}}
+                    value={threshold.windowValue}
+                    type="number"
+                    min={0}
+                    onChange={e =>
+                      editThresholdState(threshold.id, 'windowValue', e.target.value)
+                    }
+                  />
```

Comment:
putting the form into its own component with the edit state inside of it would cause fewer state changes on the entire list especially since i don't think you save more than 1 threshold at a time

Reply:
So - i _was_ thinking of implementing a "save all" button
but i just havent gotten around to that yet 😬